port/unix/omrsysinfo.c

```diff
@@ -3607,12 +3618,16 @@ readCgroupSubsystemFile(struct OMRPortLibrary *portLibrary, uint64_t subsystemFl
 	uint64_t availableSubsystem = portLibrary->sysinfo_cgroup_are_subsystems_available(portLibrary, subsystemFlag);
 
 	if (availableSubsystem != subsystemFlag) {
+		Trc_PRT_readCgroupSubsystemFile_subsystem_not_available(subsystemFlag);
 		rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_CGROUP_SUBSYSTEM_UNAVAILABLE, "cgroup subsystem %s is not available", subsystemNames[subsystem]);
 		goto _end;
 	}
 	
 	cgroup = getCgroupNameForSubsystem(portLibrary, PPG_cgroupEntryList, subsystemNames[subsystem]);
 	if (NULL == cgroup) {
+		/* If the subsystem is available and supported, cgroup must not be NULL */
+		Trc_PRT_readCgroupSubsystemFile_missing_cgroup(subsystemFlag);
+		Trc_PRT_Assert_ShouldNeverHappen();
```

Comment:
`Trc_PRT_Assert_ShouldNeverHappen` should be the last thing in the code path as it kills execution.
In this case, it should be moved to just before the `goto _end`.